### ISSUE TITLE
Unexpected NullPointerException

### ISSUE THREAD

**Issue by grnt426** [rythmengine/rythmengine#369]
I have two classes like the following:
```java
class Base {
  private name;
  // getters, setters
}

class Extended extends Base {
  private type;
  // getters, setters
}
```

I then had a template like so:

```java
@import com.my.class.Base
@import com.my.class.Extended
@args Extended myObject
// some HTML
@{p(myObject)}
@if(myObject.getType()) {  <-- NPE on this line
  // more HTML
}
```

This will produce a `NullPointerException` within the if-statement. To make sure it wasn't null, you'll notice I print out the object just before the if-statement and I get an ObjectId back from the default `.toString()` method, which should mean it definitely isn't null.

FWIW, earlier I was using Lombok, and when I did `@{p(myObject)}` I was getting a JSON version of the object with all the fields as expected. I removed Lombok from these two classes to make sure something weird wasn't going on, but I get the same `NullPointerException`.

Versions: Java8, Rythm1.1.5. Using SparkJava2.6.0 and Spring4.3.10.RELEASE

**Comment by grnt426** [comment]
I finally got around to investigating this further, and here is the problem:

It seems like the http://rythmengine.org/doc/expression#null-safe **must** be used to do anything that might contain a null. This is very surprising behavior, as it means I must litter my code with the null-expression `?` everywhere.

For example, say I provide a POJO to a template and want to print its values. I can't do the below:

```
@dao.getName()
@dao.getSomeValue()
```

As this will throw an NPE. Instead, I **must** do

```
@dao?.getName()
@dao?.getSomeValue()
```

Why isn't the default for a null value to just print nothing?

Secondly, I am uncertain how to use the "Elvis" operator linked in the next section. As of `Rythm1.2.0`, I get this exception when trying to use it, `org.rythmengine.exception.CompileException: Syntax error on token "?", delete this token`.

```
@(name ?: "No Name")
```

Lastly, even assuming I wasn't doing something wrong, and the Elvis operator did work as the documentation says, handling nulls is not simple with Rythm. If I want to print a default value, I must have blocks like this:

```
@{name = dao?.getName()}
@(name ?: "No Name")
```

While this is slightly better than having to write a full if, not being able to mix the Elvis operator and the null-expression is limiting, as I must create a third variable to store the result to use the Elvis operator.

My question is: _why does Rythm handle nulls in this very surprising way?_ The tools provided in the documentation don't help with nulls like Optional might, they seem meant to prevent Rythm from *causing* an NPE on what should have been a valid expression. Is the documentation wrong about the Elvis operator? How do I use it?

Am I just doing something completely wrong? Is my use-case for Rythm, which is a heavy reliance on POJOs rather than arrays of strings/JSON, not well-suited for Rythm?

**Comment by grnt426** [comment]
I dug through the unit tests of the project and started playing with the passing tests that use these null-expression constructs and was able to tease out what was the problem. Turns out it was on my side, as one of the objects was null within the list of objects.

However, I wasn't able to determine why I got the Syntax error above, because as I was adding proper checks/handling, the syntax problem went away. My understanding of the null-expression/Elvis operator was just wrong, so I'll close this issue.
